### PR TITLE
Only list witnesses that belong to the base machine.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: executor/src/witgen/generator.rs [hunk 1]
```diff
@@ -246,7 +255,10 @@ where
     }
 
     fn format_next_values(&self) -> Vec<String> {
-        self.format_next_values_iter(self.next.iter().enumerate())
+        self.format_next_values_iter(self.next.iter().enumerate().filter(|(i, _)| {
+            self.witnesses
```
Comment (chriseth) [inline executor/src/witgen/generator.rs:259]: I'll extract this as a function

File: executor/src/witgen/machines/machine_extractor.rs [hunk 2]
```diff
@@ -35,7 +36,7 @@ pub fn split_out_machines<'a, T: FieldElement>(
     let mut machines: Vec<Box<dyn Machine<T>>> = vec![];
 
     let all_witnesses = witness_cols.iter().map(|c| &c.poly).collect::<HashSet<_>>();
-    let mut remaining_witnesses = all_witnesses.clone();
+    let mut remaining_witnesses: HashSet<&'a PolynomialReference> = all_witnesses.clone();
```
Comment (lvella) [inline executor/src/witgen/machines/machine_extractor.rs:39]: Why did you have to specify the full type here?

If really needed, I would have done it in the previous line, either at `collect::<...>` or at `let all_witnesses: ...`, but not both.

**Review by lvella — APPROVED** [review]

File: executor/src/witgen/machines/machine_extractor.rs (hunk 2 above)
Comment (chriseth) [reply executor/src/witgen/machines/machine_extractor.rs:39]: The `'a` was the thing I mainly needed.

Comment (lvella) [reply executor/src/witgen/machines/machine_extractor.rs:39]: Still could be done only in the previous line.

Comment (chriseth) [reply executor/src/witgen/machines/machine_extractor.rs:39]: Ah, now I see! Here it is: https://github.com/powdr-org/powdr/pull/359

**Pull request merged**